src/plugins/StatusBar/StatusBar.js

```diff
@@ -106,24 +106,58 @@ module.exports = (props) => {
   }
 
   const width = typeof progressValue === 'number' ? progressValue : 100
+  const isHidden = uploadState === STATE_WAITING && !props.newFiles > 0
 
   return html`
     <div class="UppyStatusBar is-${uploadState}"
-                aria-hidden="${uploadState === STATE_WAITING}"
-                title="">
-      <progress style="display: none;" min="0" max="100" value=${progressValue}></progress>
+         aria-hidden="${isHidden}">
       <div class="UppyStatusBar-progress ${progressMode ? `is-${progressMode}` : ''}"
-           style="width: ${width}%"></div>
+           style="width: ${width}%"
+           role="progressbar"
+           aria-valuemin="0"
+           aria-valuemax="100"
+           aria-valuenow="${progressValue}"></div>
       ${progressBarContent}
+      <div class="UppyStatusBar-actions">
+        ${props.newFiles ? UploadBtn(props) : ''}
+        ${props.error ? RetryBtn(props) : ''}
+      </div>
     </div>
   `
 }
 
+const UploadBtn = (props) => {
+  return html`<button type="button"
+                      class="UppyStatusBar-actionBtn UppyStatusBar-actionBtn--upload"
+                      aria-label="${props.i18n('uploadXFiles', { smart_count: props.newFiles })}"
+                      onclick=${props.startUpload}>
+                ${props.inProgress
+                  ? props.i18n('uploadXNewFiles', { smart_count: props.newFiles })
+                  : props.i18n('uploadXFiles', { smart_count: props.newFiles })
+                }
+              </button>`
+}
+
+const RetryBtn = (props) => {
+  return html`<button type="button"
+                      class="UppyStatusBar-actionBtn UppyStatusBar-actionBtn--retry"
+                      aria-label="${props.i18n('retryUpload')}"
+                      onclick=${props.retryAll}>
+                ${props.i18n('retry')}
+              </button>`
+}
+
 const ProgressBarProcessing = (props) => {
+  const value = Math.round(props.value * 100)
   return html`
     <div class="UppyStatusBar-content">
-      ${props.mode === 'determinate' ? `${Math.round(props.value * 100)}%・` : ''}
-      ${props.message}
+      <div role="progressbar"
+           aria-valuemin="0"
+           aria-valuemax="100"
+           aria-valuenow="${value}">
```

Comment:
when `props.mode` is 'indeterminate', we should not add the `aria-valuenow` attribute

Reply:
And all of them, including min/max?

Reply:
I also think this would result in duplicated `aria-value`s, since the root element has them? Removed on `ProgressBarProcessing`.

Reply:
you can sort of do conditional attributes by using object spread, both in jsx and hyperx:

```js
html`
  <div
    ${props.mode === 'determinate'
      ? { 'aria-valuenow': value }
      : {}}
  />
`
```

but i think the setAttribute is an acceptable solution too.

Reply:
Oh, nice, I didn’t know that! Thanks!

Do we need to check for `props.mode` if we can check for `progressValue`? If `in determinate`, won’t `progressValue` be 0 or undefined? I got lost here, sorry.